Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs

```diff
@@ -116,8 +116,28 @@ public OAuthApplication ExternalLoginInformation(string refreshToken)
         /// <param name="slackEvent"></param>
         public void SlackEventUpdate(SlackEventApiAC slackEvent)
         {
-            slackEvent.Event.User.TeamId = slackEvent.TeamId;
-            _slackUserDetails.Insert(slackEvent.Event.User);
+            var user = _slackUserDetails.FirstOrDefault(x => x.UserId == slackEvent.Event.User.UserId);
+            if (user == null)
+            {
+                if (!slackEvent.Event.User.IsBot)
+                    _slackUserRepository.AddSlackUser(slackEvent.Event.User);
+            }
+        }
+
+
+        /// <summary>
+        /// Method to update slack channel table when a channel is added in team.
+        /// </summary>
+        /// <param name="slackEvent"></param>
+        public void SlackChannelAdd(SlackEventApiAC slackEvent)
+        {
+            var channel = _slackChannelDetails.FirstOrDefault(x => x.ChannelId == slackEvent.Event.Channel.ChannelId);
+            if (channel == null)
```

Comment:
Create another issue to handle the update of the channel. For now I accept this change.